### PR TITLE
Setting golang version uniformly accross repo and updating docs

### LINKED ISSUE #3023: Test and Release/Build environment should match
It is important that the test environment and release/build environment match.
For example, the same Go version should be used everywhere. 

- RPM Spec is currently specifying Go 1.11. https://github.com/openshift/odo/blob/08c339c744f8d92848b50ba816fc0a82e5cff302/rpms/openshift-odo.spec#L6
- Release build on Red Hat Brew instance is happening on 1.12.
- Travis-CI is using Go 1.13.1 https://github.com/openshift/odo/blob/08c339c744f8d92848b50ba816fc0a82e5cff302/.travis.yml#L19
- OpenShift CI is using Go 1.13 https://github.com/openshift/odo/blob/08c339c744f8d92848b50ba816fc0a82e5cff302/openshift-ci/build-root/Dockerfile#L3

Acceptance criteria:
 - [ ] use the same go versions everywhere (1.12)
 - [ ] document how to change go version (all places where it needs to be updated) 

/area release-eng
/area testing
/kind bug
/priority high

/cc @amitkrout @mohammedzee1000

### PR DESCRIPTION
**Pull request by mohammedzee1000**
- All golang version references now uniformly updated to 1.12
 - Adding all golang version references to development.adoc

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>
/kind code-refactoring
/kind documentation

**Which issue(s) this PR fixes**:

Fixes #3023

### CODE REVIEW

File: docs/dev/development.adoc [hunk 1]
```diff
@@ -9,10 +9,16 @@ toc::[]
...
 
-If that is ok, please ensure you update the required golang version, both here and in the file link:/scripts/rpm-prepare.sh[`scripts/rpm-prepare.sh`]
+If that is ok, please ensure you update the required golang version, both here and in the files below, in your PR.
+
+ * link:/scripts/rpm-prepare.sh[`scripts/rpm-prepare.sh`]
+ * link:/.travis.yml[`.travis.yml`]
+ * link:/Dockerfile.rhel[`Dockerfile.rhel`]
+ * link:/openshift-ci/build-root/Dockerfile[`openshift-ci/build-root/Dockerfile`]
+
+ First setup your fork of the odo project, following the steps below
```
Comment (dharmit) [inline docs/dev/development.adoc:21]: Please remove the additional spaces at the beginning of this line. It's causing the line to be rendered as code-block.

File: docs/dev/development.adoc [hunk 2]
```diff
@@ -9,10 +9,16 @@ toc::[]
 
 Requires *Go 1.12*
 
-**WARNING**: If you are adding any features that require a higher version of golang, such as golang 1.13
-for example, please contact maintainers to check of the releasing systems can handle the newer versions.
+**WARNING**: If you are adding any features that require a higher version of golang, that the one mentioned above, please contact maintainers to check of the releasing systems can handle the newer versions.
```
Comment (dharmit) [inline docs/dev/development.adoc:12]: Can you put the stuff that user needs to be warned about in an "Admonition" block? Please refer https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#admon-bl.

File: docs/dev/development.adoc (hunk 1 above)
Comment (mohammedzee1000) [reply docs/dev/development.adoc:21]: Done

File: docs/dev/development.adoc (hunk 2 above)
Comment (mohammedzee1000) [reply docs/dev/development.adoc:12]: Done, thanks

Comment (mohammedzee1000) [reply docs/dev/development.adoc]: Interesting looks like this does not render in normal view on github but raw view shows it. Il add back the `warning` as well to be sure

Comment (dharmit) [reply docs/dev/development.adoc]: It's still not rendering well when I look at the rich diff. The list of links is outside the warning box. Here's one way you could do it:
```
[WARNING]
====
If you are adding any features that require a higher version of golang, that the one mentioned above, please contact maintainers to check of the releasing systems can handle the newer versions. If that is ok, please ensure you update the required golang version, both here and in the files below, in your PR.

.List of files to update for golang version
 * link:/scripts/rpm-prepare.sh[`scripts/rpm-prepare.sh`]
 * link:/.travis.yml[`.travis.yml`]
 * link:/Dockerfile.rhel[`Dockerfile.rhel`]
 * link:/openshift-ci/build-root/Dockerfile[`openshift-ci/build-root/Dockerfile`]
====
```

File: docs/dev/development.adoc [hunk 3]
```diff
@@ -9,10 +9,21 @@ toc::[]
...
 
-**WARNING**: If you are adding any features that require a higher version of golang, such as golang 1.13
-for example, please contact maintainers to check of the releasing systems can handle the newer versions.
+Testing and release builds happen with the above version. Developers are advised to stick to this version if they can but it is not compulsory.
 
-If that is ok, please ensure you update the required golang version, both here and in the file link:/scripts/rpm-prepare.sh[`scripts/rpm-prepare.sh`]
+
+[WARNING]
+====
+If you are adding any features that require a higher version of golang, that the one mentioned above, please contact maintainers to check of the releasing systems can handle the newer versions. If that is ok, please ensure you update the required golang version, both here and in the files below, in your PR.
```
Comment (cdrage) [inline docs/dev/development.adoc:17]: `than the one mentioned above`

Comment (cdrage) [reply docs/dev/development.adoc:17]: `please contact the maintainers`

Comment (cdrage) [reply docs/dev/development.adoc:17]: `please contact the maintainers in order to check if the releasing systems can handle the newer versions`

Comment (mohammedzee1000) [reply docs/dev/development.adoc]: fixed

**Pull request merged**